client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/settings.js

```diff
@@ -0,0 +1,354 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import CompactCard from 'components/card/compact';
+import Dialog from 'components/dialog';
+import Dropdown from 'woocommerce/woocommerce-services/components/dropdown';
+import Checkbox from 'woocommerce/woocommerce-services/components/checkbox';
+import TextField from 'woocommerce/woocommerce-services/components/text-field';
+import {
+	getDestinationCountryNames,
+	getStateNames,
+} from 'woocommerce/woocommerce-services/state/shipping-label/selectors';
+import {
+	getCarrierAccountsState,
+	getFormErrors,
+	getFormValidState,
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/selectors';
+import {
+	setVisibilityCancelConnectionDialog,
+	submitCarrierSettings,
+	updateCarrierSettings,
+	toggleShowUPSInvoiceFields,
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/actions';
+import { getCountryName } from 'woocommerce/state/sites/data/locations/selectors';
+import { decodeEntities } from 'lib/formatting';
+
+const CarrierAccountSettings = props => {
+	const {
+		siteId,
+		carrier,
+		countryNames,
+		fieldErrors,
+		values,
+		stateNames,
+		isFormValid,
+		showUPSInvoiceFields,
+		showCancelConnectionDialog,
+		translate,
+	} = props;
+
+	const getValue = fieldName => {
+		return values[ fieldName ] ? decodeEntities( values[ fieldName ] ) : '';
+	}
+	const updateValue = fieldName => newValue =>
+		props.updateCarrierSettings( siteId, carrier, fieldName, newValue );
+	const submitCarrierSettingsHandler = () =>
+		props.submitCarrierSettings( siteId, carrier );
+	const showCancelDialogHandler = () => {
+		props.setVisibilityCancelConnectionDialog( siteId, carrier, true );
+	}
+	const hideCancelDialogHandler = () => {
+		props.setVisibilityCancelConnectionDialog( siteId, carrier, false );
+	}
+	const updateShowUPSInvoiceFields = () => {
+		props.toggleShowUPSInvoiceFields( siteId, carrier );
```

Comment:
Show/hide UPS forms seems to be inside only `CarrierAccountSettings` component, also curious why redux is preferred over local state? I always go with local state before putting things in redux, so I am actually interested to know if there are something I am not seeing.

Reply:
See me answer above! :smile: